nomad/job_endpoint_hook_connect.go

```diff
@@ -477,12 +501,12 @@ func newConnectGatewayTask(prefix, service string, netHost, customizedTls bool)
 	}
 }
 
-func newConnectSidecarTask(service string) *structs.Task {
+func newConnectSidecarTask(service, driver string) *structs.Task {
 	return &structs.Task{
 		// Name is used in container name so must start with '[A-Za-z0-9]'
 		Name:          fmt.Sprintf("%s-%s", structs.ConnectProxyPrefix, service),
 		Kind:          structs.NewTaskKind(structs.ConnectProxyPrefix, service),
-		Driver:        "docker",
+		Driver:        driver,
 		Config:        connectSidecarDriverConfig(),
```

Comment:
Would it be worth adding a docsting to `connectSidecarDriverConfig()` to note that it should be compatible with Docker _and_ Podman? 

Or maybe even pass the driver choice here. It can be ignored in the current implementation, but may be handy if there are conflicting configs between the drivers or we add support for other plugins.

Reply:
Added a comment; with a note about passing in the parameter one day if needed